### PR TITLE
Add support for the Reporting resources

### PR DESCRIPTION
**Pull request by remi-stripe**
r? @ob-stripe 
cc @stripe/api-libraries @brahn-stripe

### CODE REVIEW

File: src/main/java/com/stripe/net/ApiResource.java [hunk 1]
```diff
@@ -85,6 +85,8 @@ private static String className(Class<?> clazz) {
       className = "issuing/" + className;
     } else if (clazz.getName().contains("com.stripe.model.sigma.")) {
       className = "sigma/" + className;
+    } else if (clazz.getName().contains("com.stripe.model.reporting.")) {
+      className = "reporting/" + className;
```
Comment (remi-stripe) [inline src/main/java/com/stripe/net/ApiResource.java]: I know we need to fix this but I think this PR is the wrong place. Though we should schedule this work soon (happy to take, we just need to sync on it a bit)

File: src/main/java/com/stripe/model/reporting/ReportRun.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+@Getter
+@Setter
+@EqualsAndHashCode(callSuper = false)
+public class ReportRun extends ApiResource implements HasId {
+  @Getter(onMethod = @__({@Override})) String id;
+  String object;
+  Long created;
+  String error;
+  Boolean livemode;
```
Comment (brandur-stripe) [inline src/main/java/com/stripe/model/reporting/ReportRun.java]: I think that these actually don't have a livemode property :/

File: src/main/java/com/stripe/model/reporting/ReportType.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+@Getter
+@Setter
+@EqualsAndHashCode(callSuper = false)
+public class ReportType extends ApiResource implements HasId {
+  @Getter(onMethod = @__({@Override})) String id;
+  String object;
+  Long dataAvailableEnd;
+  Long dataAvailableStart;
+  String name;
+  Long update;
```
Comment (brandur-stripe) [inline src/main/java/com/stripe/model/reporting/ReportType.java]: I think this one is "updated".

File: src/main/java/com/stripe/model/reporting/ReportRun.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      ReportRun.class, options);
+  }
+  // </editor-fold>
+
+  @Getter
+  @Setter
+  @EqualsAndHashCode(callSuper = false)
+  public final class Parameters extends StripeObject {
+    Long intervalEnd;
+    Long intervalStart;
```
Comment (brandur-stripe) [inline src/main/java/com/stripe/model/reporting/ReportRun.java]: I'm seeing quite a few additional fields as part of this one — is the idea to leave them out for now?

image

Comment (ob-stripe) [inline src/main/java/com/stripe/model/reporting/ReportRun.java]: There are some other attributes that are missing here:
```java
String connectedAccount;
String currency;
String payout;
String reportingCategory;
```

File: src/main/java/com/stripe/model/reporting/ReportRun.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  public static ReportRunCollection list(Map<String, Object> params, RequestOptions options)
+      throws StripeException {
+    return requestCollection(classUrl(ReportRun.class), params,
+      ReportRunCollection.class, options);
+  }
+  // </editor-fold>
+
+  // <editor-fold desc="retrieve">
+  /**
+   * Retrieve an report run.
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/reporting/ReportRun.java]: Should be "Retrieve a report run."

File: src/main/java/com/stripe/model/reporting/ReportRun.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+   */
+  public static ReportRun retrieve(String id) throws StripeException {
+    return retrieve(id, null);
+  }
+
+  /**
+   * Retrieve an report run.
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/reporting/ReportRun.java]: Ditto.

File: src/main/java/com/stripe/model/reporting/ReportRun.java [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+   */
+  public static ReportRun retrieve(String id, RequestOptions options) throws StripeException {
+    return retrieve(id, null, options);
+  }
+
+  /**
+   * Retrieve an report run.
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/reporting/ReportRun.java]: Ditto.

File: src/main/java/com/stripe/model/reporting/ReportType.java (hunk 3 above)
Comment (ob-stripe) [inline src/main/java/com/stripe/model/reporting/ReportType.java]: Should be `Long updated`.

File: src/main/java/com/stripe/model/reporting/ReportType.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+  public static ReportTypeCollection list(Map<String, Object> params, RequestOptions options)
+      throws StripeException {
+    return requestCollection(classUrl(ReportType.class), params,
+      ReportTypeCollection.class, options);
+  }
+  // </editor-fold>
+
+  // <editor-fold desc="retrieve">
+  /**
+   * Retrieve an report type.
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/reporting/ReportType.java]: Should be "Retrieve a report type."

File: src/main/java/com/stripe/model/reporting/ReportType.java [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+   */
+  public static ReportType retrieve(String id) throws StripeException {
+    return retrieve(id, null);
+  }
+
+  /**
+   * Retrieve an report type.
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/reporting/ReportType.java]: Ditto.

File: src/main/java/com/stripe/model/reporting/ReportType.java [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+   */
+  public static ReportType retrieve(String id, RequestOptions options) throws StripeException {
+    return retrieve(id, null, options);
+  }
+
+  /**
+   * Retrieve an report type.
```
Comment (ob-stripe) [inline src/main/java/com/stripe/model/reporting/ReportType.java]: Ditto.

File: src/main/java/com/stripe/net/ApiResource.java (hunk 1 above)
Comment (ob-stripe) [reply src/main/java/com/stripe/net/ApiResource.java]: 👍

File: src/main/java/com/stripe/model/reporting/ReportRun.java (hunk 2 above)
Comment (remi-stripe) [reply src/main/java/com/stripe/model/reporting/ReportRun.java]: The `ReportRun` resource does. It was added a few hours ago as I asked for it

**Comment by brandur-stripe** [comment]
Oops, I reviewed the wrong pull request, haha.

**Comment by ob-stripe** [comment]
Oops, looks like @brandur-stripe was faster :)

File: src/main/java/com/stripe/model/reporting/ReportRun.java (hunk 4 above)
Comment (remi-stripe) [reply src/main/java/com/stripe/model/reporting/ReportRun.java]: Ah yes that's what I get for trusting the fixture in the docs 🤦‍♂️, fixing.

File: src/main/java/com/stripe/model/reporting/ReportRun.java (hunk 2 above)
Comment (brandur-stripe) [reply src/main/java/com/stripe/model/reporting/ReportRun.java]: Ah you're right. I see it in there now.

**Comment by remi-stripe** [comment]
Okay fixed all the issues. Thank you both for catching them before I did go and .net incorrectly :p

**Comment by ob-stripe** [comment]
Released as 6.11.0.

**Pull request merged**